test/integration/features/coolstore.feature

```diff
@@ -0,0 +1,137 @@
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift addons enable xpaas"
+     Then stdout should contain
+      """
+      Addon 'xpaas' enabled
+      """
+
+  Scenario: Starting Minishift
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift start --memory 4096 --iso-url centos" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: User can create new namespace coolstore
+   Given Minishift has state "Running"
+    When executing "oc new-project coolstore" succeeds
+    Then stdout should contain
+     """
+     Now using project "coolstore"
+     """
+  
+  Scenario: User can create a template for coolstore 
+   Given Minishift has state "Running"
+    When executing "oc create -f $(template_dir)/coolstore.yaml" succeeds
+    Then stdout should contain
+     """
+     template "coolstore" created
+     """
+
+  Scenario: User can deploy the coolstore template to namespace coolstore 
+   Given Minishift has state "Running"
+    When executing "oc new-app --template=coolstore" succeeds
+    Then stdout should contain
+     """
+     Success
+     """
+
+  Scenario: Successfully deployed web-ui service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig web-ui --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed inventory service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig inventory --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed catalog service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig catalog --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed cart service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig cart --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully rolled out gateway to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig coolstore-gw --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed all services
+   Given Minishift has state "Running" 
+    When executing "oc status -v" succeeds
+    Then stdout should not contain
+     """
+     has failed
+     """
+
+  Scenario: Should be able to get the pod identifier for the API gateway 
+   Given Minishift has state "Running" 
+    When set variable "coolstore-gw" executing "oc get pods -o name -l application=coolstore-gw"
```

Comment:
@agajdosi uses environment variables as:
```
// steps for environmental variables
s.Step(`^user sets environmental variable "([^"]*)" to "([^"]*)"$`, setEnvVariable)
s.Step(`^user unsets environmental variable "([^"]*)"$`, unsetEnvVariable)
```

which in a similar fashions set the a record by name with the result from another command. This is also my intention here... I think for consistency it is best to use the pattern [variable] [from], but perhaps dropping the 'user' in the phrasing